tutorials/throttle/src/main/java/com/datatorrent/examples/throttle/ThrottlingStatsListener.java

```diff
@@ -26,8 +26,13 @@
     // Slowdown input if the window difference between operators increases beyond this value
     long maxThreshold = 100;
     // restore input operator to normal speed if the window difference falls below this threshold
-    long minThreshold = 100;;
-    boolean normalState = true;
+    long minThreshold = 100;
+
+    Map<Integer, ThrottleState> throttleStates = Maps.newHashMap();
```

Comment:
This "out-of-band" communication among operators is rather subtle: suggest helping out the user by adding a comment to the `processStats()` method stating that it is executed by the AppMaster once per second; this complements the existing comment on the `InputSlowdownRequest` class.

Couple of other thoughts, maybe for the future:

1. The loop to find max/min can be avoided by keeping track of the max/min dynamically on each call; the `currentWindowIds` map would no longer be needed.

2. We can determine if the current operator is the (assuming there is only one which is the case here) input operator
by  checking if  `batchedOperatorStats.getLastWindowedStats().inputPorts`  is null or empty; for all other operators
we can simply send the null response. This then avoids the need for the `instanceof` check in `execute()`.